src/pages/Detail/Detail.page.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from 'react';
+import { useLocation } from 'react-router';
+import { escapeString } from '../../utils/escapeString';
+import { formatDate } from '../../utils/formatDate';
+import { HomeLink, Detail as DetailWrap, RelatedCardWrapper, RelatedCard, Video, IframeWrapper, Desc, DateSpan, Title } from './styled';
+import mock from '../../mockRelatedVideos.json';
+
+function Detail() {
+    const location = useLocation();
+    const [relatedVideos, setRelatedVideos] = useState(mock.items);
+    const { videoId, title, description, publishedAt } = location.state;
+    const [video, setVideo] = useState({ videoId, title, description, publishedAt});
+    const baseApiUrl = 'https://youtube.googleapis.com/youtube/v3/search';
+    
+    useEffect(() => {
+      const getRelated = async () => {
+          let response = await fetch(`${baseApiUrl}?part=snippet&relatedToVideoId=${videoId}&type=video&key=${process.env.REACT_APP_API_KEY}`)
+  
+          if (response.ok) { // if HTTP-status is 200-299
+            let json = await response.json();
+            setRelatedVideos(json.items);
+          } else {
+            alert(`"HTTP-Error: " + ${response.status}\nGet Related Videos: Failed\nProbably because API_KEY is not set in your env`);
+          }
+      };
```

Comment:
A couple of comments, I like that you are using variables to create the url, this makes the code easy to maintain 👍. This function could be in its own file as a custom hook, this way you could reuse it and also test it easily. And finally, if this was meant to be used by non-developers I would change the `alert` for a console log